cluster/cluster.go

```diff
@@ -17,20 +19,54 @@ var (
 
 type Cluster struct {
 	sync.RWMutex
+	store           *state.Store
 	tlsConfig       *tls.Config
 	eventHandlers   []EventHandler
 	nodes           map[string]*Node
 	overcommitRatio float64
 }
 
-func NewCluster(tlsConfig *tls.Config, overcommitRatio float64) *Cluster {
+func NewCluster(store *state.Store, tlsConfig *tls.Config, overcommitRatio float64) *Cluster {
 	return &Cluster{
 		tlsConfig:       tlsConfig,
 		nodes:           make(map[string]*Node),
+		store:           store,
 		overcommitRatio: overcommitRatio,
 	}
 }
 
+// Deploy a container into a `specific` node on the cluster.
+func (c *Cluster) DeployContainer(node *Node, config *dockerclient.ContainerConfig, name string) (*Container, error) {
+	container, err := node.Create(config, name, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// Commit the requested state.
+	st := &state.RequestedState{
```

Comment:
Btw, looking at this, I wonder if go is flexible enough to serialize fields that is missing when it reads json from disk?
The reason I'm asking is since we're serializing into json from a go struct, this RequestedState obj might change in schema and we need the store to handle this.